src/toast/todmap/sim_tod.py

```diff
@@ -1013,140 +1018,166 @@ def scan_range(self):
-            offset_out = max(0, i)
-            ngood = nstep - offset_in
-            if ngood > 0:
-                self._commonflags[
-                    offset_out : offset_out + ngood
-                ] |= self.LEFTRIGHT_TURNAROUND
-                ii = np.arange(offset_in, offset_in + ngood)
-                self._az[offset_out : offset_out + ngood] = (
-                    az_last + ii * dazdt - 0.5 * scan_accel * ii ** 2
-                )
-                halfway = i + nstep_full // 2
-                if halfway > 0 and halfway < samples:
-                    starts.append(halfway)
-            i += nstep
-            if i == samples:
-                break
-            az_last += dazdt * nstep - 0.5 * scan_accel * nstep ** 2
-            #
-            # Right to left, fixed rate
-            #
-            self._stable_starts.append(i)
-            dazdt = -scanrate
-            nstep = min(int((lim_left - az_last) // dazdt) + 1, samples - i)
-            offset_in = max(0, -i)
-            offset_out = max(0, i)
-            ngood = nstep - offset_in
-            if ngood > 0:
-                self._commonflags[
-                    offset_out : offset_out + ngood
-                ] |= self.RIGHTLEFT_SCAN
-                self._az[offset_out : offset_out + ngood] = (
-                    az_last + np.arange(offset_in, offset_in + ngood) * dazdt
-                )
-            i += nstep
-            self._stable_stops.append(i)
-            if i == samples:
-                break
-            az_last += dazdt * nstep
-            #
-            # Right to left, turnaround
-            #
-            nstep_full = int((2 * scanrate) // scan_accel) + 1
-            nstep = min(int(nstep_full), samples - i)
-            offset_in = max(0, -i)
-            offset_out = max(0, i)
-            ngood = nstep - offset_in
-            if ngood > 0:
-                self._commonflags[
-                    offset_out : offset_out + ngood
-                ] |= self.RIGHTLEFT_TURNAROUND
-                ii = np.arange(offset_in, offset_in + ngood)
-                self._az[offset_out : offset_out + ngood] = (
-                    az_last + ii * dazdt + 0.5 * scan_accel * ii ** 2
+        """ Simulate a constant elevation scan, either constant rate or
+        1/sin(az)-modulated.
+
+        """
+
+        # Begin by simulating one full scan with turnarounds.
+        # It will be used to interpolate the full CES.
+        # `nstep` is the number of steps used for one sweep or one
+        # turnaround.
+
+        nstep = 10000
+
+        azmin, azmax = [self._azmin, self._azmax]
+        if self._sinc_modulation:
+            # We always simulate a rising sinc scan and then
+            # mirror it if necessary
+            azmin %= np.pi
+            azmax %= np.pi
+            if azmin > azmax:
+                raise RuntimeError(
+                    "Cannot scan across zero meridian with sinc-modulated scan"
```

Comment:
How much of a limitation is this?

Reply:
It is a fundamental limitation of the sinc-modulated strategy. 1/sin(az) diverges on the zero meridian so one cannot scan across.